employee_api/handlers/employee.go

```diff
@@ -0,0 +1,246 @@
+package employee_handler
+
+import (
+	"HR-system/employee_api/models"
+	responses "HR-system/employee_api/utils"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/gorilla/mux"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+type Route struct {
+	Name        string
+	Method      string
+	Pattern     string
+	HandlerFunc http.HandlerFunc
+}
+
+type Server struct {
+	DB     *gorm.DB
+	Router *mux.Router
+	Routes []Route
+}
+
+func (s *Server) Run(
+	host, port, user, dbname, password, sslmode string,
+) {
+	s.initializeDB(host, port, user, dbname, password, sslmode)
+	s.startServe(":8080")
+}
+
+func (s *Server) collectRoutes() []Route {
+	return []Route{
+		{
+			"Index",
+			"GET",
+			"/",
+			s.index,
+		},
+		{
+			"Create employee",
+			"POST",
+			"/api/v1/employee/",
+			s.create,
+		},
+		{
+			"Get employee by id",
+			"GET",
+			"/api/v1/employee/{id}",
+			s.get,
+		},
+		{
+			"Update employee",
+			"PUT",
+			"/api/v1/employee/{id}",
+			s.update,
+		},
+		{
+			"Delete employee",
+			"DELETE",
+			"/api/v1/employee/{id}",
+			s.delete,
+		},
+	}
+
+}
+
+func (server *Server) startServe(addr string) {
+	log.Println("Listening to port 8080")
+	log.Fatal(http.ListenAndServe(addr, server.Router))
+}
+
+func (s *Server) initializeDB(host, port, user, dbname, password, sslmode string) {
+	databaseUrl := fmt.Sprintf("host=%s port=%s user=%s dbname=%s password=%s sslmode=%s", host, port, user, dbname, password, sslmode)
+	db, err := gorm.Open(postgres.Open(databaseUrl), &gorm.Config{})
+
+	if err != nil {
+		log.Printf("Cannot connect to %s database", dbname)
+		log.Fatal("This is the error:", err)
+		return
+	}
+
+	s.DB = db
+
+	s.DB.AutoMigrate(&models.Employee{})
+
+	s.Router = mux.NewRouter()
+	s.Routes = s.collectRoutes()
+	for _, route := range s.Routes {
+		s.Router.HandleFunc(route.Pattern, route.HandlerFunc).Methods(route.Method)
+	}
+}
+
+func (s *Server) create(w http.ResponseWriter, r *http.Request) {
+	// read data from request body
+	body, err := io.ReadAll(r.Body)
```

Comment:
I suggest use `gin`, maybe it will be fast and easy to use.

Reply:
OK, I change to gin framework now